core/src/test/java/org/elasticsearch/validate/SimpleValidateQueryIT.java

```diff
@@ -252,6 +255,35 @@ public void testExplainWithRewriteValidateQuery() throws Exception {
                 containsString("field:huge field:pidgin"), true);
     }
 
+    public void testExplainWithRewriteValidateQueryAllShards() throws Exception {
+        client().admin().indices().prepareCreate("test")
+            .addMapping("type1", "field", "type=text,analyzer=whitespace")
+            .setSettings(SETTING_NUMBER_OF_SHARDS, 2).get();
+        client().prepareIndex("test", "type1", "1")
```

Comment:
I'd leave a comment here about how you are relying on specific routing behaviors for the result to be right.